### PR TITLE
chore(sdk): optimizing `bucket.exists()` for aws targets

### PR DESCRIPTION
**Pull request by garysassano**
I was randomly thinking that using `ListObjectsV2Command` may not be the most efficient way to check if an object exists in a S3 bucket, since you are asking AWS to look for all the objects that have a specific prefix even though the API returns only one object when filtering with `MaxKeys: 1`. I thought that probably just checking for the metadata of the single object using `HeadObjectCommand` would be enough to determine whether the object exists.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by garysassano** [comment]
Not sure if this optimization is overkill to be honest, and it would require to update also the tests using mock AWS resources.

File: libs/wingsdk/src/shared-aws/bucket.inflight.ts [hunk 1]
```diff
@@ -27,13 +29,13 @@ export class BucketClient implements IBucketClient {
    * @param key Key of the object
    */
   public async exists(key: string): Promise<boolean> {
-    const command = new ListObjectsV2Command({
+    const command = new HeadObjectCommand({
```
Comment (tsuf239) [inline libs/wingsdk/src/shared-aws/bucket.inflight.ts:32]: Nice! That's a great initiative 💫 
Can you please replace this in the tests as well? 
(`wing/libs/wingsdk/test/shared-aws/bucket.inflight.test.ts`)

**Review by Chriscbr — APPROVED** [review]

File: libs/wingsdk/src/shared-aws/bucket.inflight.ts (hunk 1 above)
Comment (Chriscbr) [inline libs/wingsdk/src/shared-aws/bucket.inflight.ts:32]: Do the AWS permissions need to be updated at all? see here https://github.com/winglang/wing/blob/f8a5abf8a7a6a143f3e089d1fbe2c2ef47904bfc/libs/wingsdk/src/shared-aws/permissions.ts#L84

Comment (garysassano) [reply libs/wingsdk/src/shared-aws/bucket.inflight.ts:32]: Yes, in order to retrieve the object's metadata, the `HeadObjectCommand` requires also the `s3:GetObject` and `s3:GetObjectAttributes` permissions. Thanks for reminding me, I've fixed it now.

**Pull request merged**